cpp/src/parquet/column_writer.cc

```diff
@@ -1382,11 +1383,15 @@ class TypedColumnWriterImpl : public ColumnWriterImpl,
                                          chunk.levels_to_write, *chunk_array, ctx,
                                          maybe_parent_nulls));
         }
-        if (num_buffered_values_ > 0) {
+        bool is_last_chunk = i == chunks.size() - 1;
+        if (num_buffered_values_ > 0 && !is_last_chunk) {
           // Explicitly add a new data page according to the content-defined chunk
           // boundaries. This way the same chunks will have the same byte-sequence
           // in the resulting file, which can be identified by content addressible
           // storage.
+          // Note that the last chunk doesn't trigger a new data page in order to
+          // allow subsequent WriteArrow() calls to continue writing to the same
+          // data page, the chunker's state is not being reset after the last chunk.
```

Comment:
Actually, I mean, is there two cases?
1. Chunker should end at the final value
2. Chunker should wait for more values

Should we distinct these two? Or last chunk is always not written?

Reply:
Case 2.

When we feed the last chunk through the chunker, it doesn't detect a boundary and it doesn't exceed the maximum size, so its state is not being reset. We can continue to feed values through it and it will continue the chunk calculation from that point. 
The writer has a partial chunk at the end so we don't need to explicitly call `AddDataPage()`. The user will either call `WriteArrow()` again to continue writing the chunk, or the writer will be closed which flushes the currently active page.

Reply:
Assume a scenerio, when feed last chunk, what if `["a", ..., long-final string]`, long-final string's size exceeds maximum chunk size. What would happen when long-final string is solely in final chunk and not being flushed?